add geom_corrosion_range function

libbymckenna opened the pull request:
Add in a function that draws rectangles for ideal corrosion/scaling ranges. Also just testing out how forking/contributing works. 

Closes #2

libbymckenna committed:
add geom_corrosion_range function

libbymckenna committed:
typo

libbymckenna commented:
Ranges from here. For QC, suggest using Gilbert code, line 298.
<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:dt="uuid:C2F41010-65B3-11d1-A29F-00AA00C14882"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=OneNote.File>
<meta name=Generator content="Microsoft OneNote 15">
</head>

<body lang=en-US style='font-family:Calibri;font-size:11.0pt'>
<!--StartFragment-->

<div style='direction:ltr;border-width:100%'>

<div style='direction:ltr;margin-top:0in;margin-left:0in;width:5.1583in'>

<div style='direction:ltr;margin-top:0in;margin-left:0in;width:5.1583in'>

<div style='direction:ltr'>


Table 1. Summary of corrosion   and scaling indices |   |   |   |  
-- | -- | -- | -- | --
Scaling Index | Description | Non-Scale Forming | Neutral | Scale Forming
Langelier Saturation Index   (LSI) | Indicates whether CaCO3 scaling will or will not occur, rather         than corrosivity.      Used to determine feasible pH adjustment         without scaling. | < 0 | -- | > 0
Ryznar Index (RI) | Indicator of scaling potential, rather         than corrosivity.     Mild steel corrosion may occur at RI         > 8.5 | >6.8 | 6.2 – 6.8 | <6.3
Calcium carbonate precipitation potential   (CCPP) | Predicts the mass of CaCO3 that will precipitate as the water         comes to equilibrium with CaCO3.     Measurement of scale formation, rather         than corrosivity. | <-5 | -5 – 10 | >10
Corrosion Index | Description | Corrosive | Marginally Corrosive | Non-Corrosive
Aggressiveness Index | Developed to measure corrosivity in         asbestos cement pipes.     Not appropriate indicator for systems         without asbestos cement pipes. | <10 | 10 – 11.9 | >12
Chloride-to-sulfate mass ratio   (CSMR) | Indicator of galvanic corrosion     For lead solder pipe joints     Less applicable with alkalinity > 50         mg/L as CaCO3. | > 0.5 | 0.2 - 0.5 | < 0.2
Larson-Skold Index   (LI) | Describes the corrosivity of water         towards mild steel     Not accurate for very low or very high         alkalinity waters.     Developed for Great Lakes water quality. | > 1.2 | 0.8 - 1.2 | < 0.8



</div>

<p style='margin:0in;font-family:Calibri;font-size:11.0pt'>&nbsp;</p>

libbymckenna committed:
update index ranges

sierrajohnson commented on R/Plot_Tools.R:
```diff
@@ -144,14 +144,52 @@ bc_pal <- function(palette = "primary", reverse = FALSE, ...) {
 
-########################################################################################################################*
+# GGPLOT CORROSION FUNCTIONS ----
+#' BC Color Palette with scale_color
```
```suggestion
#' Automatically add typical corrosion ranges to plots
```

sierrajohnson commented on R/Plot_Tools.R:
```diff
@@ -144,14 +144,52 @@ bc_pal <- function(palette = "primary", reverse = FALSE, ...) {
+#' @param data Data frame containing a column of named corrosion/scaling indices
+#' @param alpha Set the opacity of the rectangles, default is 0.1
+#' @param fill Set the color of the rectangles, default is "cyan"
+#' @param index_column Name of the columns with corrosion/scaling indices
```
Does this work with more than one column? If not, suggested language change below.
```suggestion
#' @param index_column Name of the column in the plot data frame with corrosion/scaling indices names (should be the column used for facetting)
```

sierrajohnson commented on R/Plot_Tools.R:
```diff
@@ -144,14 +144,52 @@ bc_pal <- function(palette = "primary", reverse = FALSE, ...) {
+# GGPLOT CORROSION FUNCTIONS ----
+#' BC Color Palette with scale_color
+#' @param data Data frame containing a column of named corrosion/scaling indices
+#' @param alpha Set the opacity of the rectangles, default is 0.1
```
```suggestion
#' @param alpha Set the opacity of the rectangles, default is 0.3
```

libbymckenna committed:
Apply suggestions from code review

Co-authored-by: Sierra Johnson <80361498+sierrajohnson@users.noreply.github.com>

libbymckenna committed:
Hupdate rectange ranges for indices based on this doc from Damon:
https://brwncald-my.sharepoint.com/:w:/p/lmckenna/Ebhc5hMxdZpHq5aKyoaGijQBheBR0EYLVtEEMJFaXpnARw?e=kRv8dX

libbymckenna committed:
Merge branch 'corrosion_ranges' of https://github.com/libbymckenna/bctools into corrosion_ranges

libbymckenna committed:
Add example, add file where ranges came from
